diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76ba69b..6a03033 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,8 +77,9 @@ if(NOT nlohmann_json_POPULATED)
     add_subdirectory(${nlohmann_json_SOURCE_DIR} ${nlohmann_json_BINARY_DIR})
 endif()
 
-target_include_directories(faiss PUBLIC ${nlohmann_json_SOURCE_DIR}/include)
+target_include_directories(faiss PUBLIC
+  $<BUILD_INTERFACE:${nlohmann_json_SOURCE_DIR}/include>
+)
 
 find_package(ZLIB REQUIRED)
 target_link_libraries(faiss PUBLIC ${ZLIB_LIBRARIES})
-
